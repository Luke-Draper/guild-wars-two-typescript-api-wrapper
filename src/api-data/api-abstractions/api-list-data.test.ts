import MockAPIListData from "./__mock__/api-list-data";

describe("APIListData", () => {
	it("instantiates", () => {
		expect(new MockAPIListData([1, 2, 3])).toBeInstanceOf(MockAPIListData);
	});
	it("throws error on incorrect information", () => {
		expect(() => {
			return new MockAPIListData({ id: 1 });
		}).toThrowErrorMatchingSnapshot();
	});
	it("throws error on incorrect information in array", () => {
		expect(() => {
			return new MockAPIListData([1, 2, 3, "fail"]);
		}).toThrowErrorMatchingSnapshot();
	});
	it("throws error on no information", () => {
		expect(() => {
			return new MockAPIListData({});
		}).toThrowErrorMatchingSnapshot();
	});
});
