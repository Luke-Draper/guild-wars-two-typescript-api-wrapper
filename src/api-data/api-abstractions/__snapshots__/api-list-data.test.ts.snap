// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APIListData throws error on incorrect information 1`] = `"Invalid Data input to APIListData constructor"`;

exports[`APIListData throws error on incorrect information in array 1`] = `"Invalid Data input to APIListData constructor"`;

exports[`APIListData throws error on no information 1`] = `"Invalid Data input to APIListData constructor"`;
